lib/vector-core/src/config/global_options.rs

```diff
@@ -39,8 +39,6 @@ pub struct GlobalOptions {
     pub timezone: TimeZone,
     #[serde(skip_serializing_if = "crate::serde::skip_serializing_if_default")]
     pub proxy: ProxyConfig,
-    #[serde(skip)]
-    pub enterprise: bool,
```

Comment:
Previously, this was used to control setting DD request headers in the `datadog_logs` sink. However, by coupling enterprise configuration with global options, we prevent reloads that involve adding/removing the `[enterprise]` configuration section. 

The coupling/global option seem unnecessary, so this was refactored into a hidden config option directly on `datadog_logs` sink.